src/main/java/com/puppycrawl/tools/checkstyle/checks/OrderedPropertiesCheck.java

```diff
@@ -0,0 +1,227 @@
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.Properties;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck;
+import com.puppycrawl.tools.checkstyle.api.FileText;
+
+/**
+ * This implementation checks that keys of properties files are sorted.
+ * It is using the binary order from String.CASE_INSENSITIVE_ORDER.compare()
+ * The sort direction is ascending.
+ *
+ *   <p>
+ *       Example properties file:
+ *   </p>
+ *   <pre>
+ *       &#64;=64 (it is the 'at' sign)
+ *       A=65
+ *       a=97
+ *       key=107 than nothing
+ *       key.sub=k is 107 and dot is 46
+ *   </pre>
+ *   <ul>
+ *       <li>
+ *           The &#64; sign is on position 64 and comes first.
+ *       </li>
+ *       <li>
+ *           The capital A is on 65 and the lowercase a is on position 97 on the ascii table.
+ *       </li>
+ *       <li>
+ *           Key and subkey are in correct order here, because only keys are relevant.
+ *           Therefore on line 5 you have only "key" an nothing behind.
+ *           On line 6 you have "key." The dot is on position 46 which is higher than nothing.
+ *       </li>
+ *   </ul>
+ * Known limitation: The key should not contain a newline.
+ * The string compare will work, but not the line number reporting.
+ */
+@StatelessCheck
+public class OrderedPropertiesCheck extends AbstractFileSetCheck {
+
+    /**
+     * Localization key for check violation.
+     */
+    public static final String MSG_KEY = "properties.notSorted.property";
+    /**
+     * Localization key for IO exception occurred on file open.
+     */
+    public static final String MSG_IO_EXCEPTION_KEY = "unable.open.cause";
+    /**
+     * Pattern matching single space.
+     */
+    private static final Pattern SPACE_PATTERN = Pattern.compile(" ");
```

Comment:
please explain this pattern, such simple pattern usually do not regexp.

Reply:
This space pattern is used to construct a key pattern to search for keys in FileText. This pattern was taken from UniquePropertiesCheck and works well.